policy-test/tests/accepted_http_route_status.rs

```diff
@@ -0,0 +1,421 @@
+        )
+        .await;
+        // Wait until route is updated with a status
+        let parent_status = await_route_status(&client, &ns, "test-multiple-parents-route")
+            .await
+            .parents;
+
+        // Find status for invalid parent and extract the condition
+        let invalid_cond = parent_status
+            .iter()
+            .find(|route_status| route_status.parent_ref.name == "test-invalid-server")
+            .expect("must have a route status set for invalid server")
+            .conditions
+            .clone()
+            .into_iter()
+            .find(|cond| cond.type_ == "Accepted")
+            .expect("must have at least one 'Accepted' condition set for invalid parent");
+        // Route shouldn't be accepted
+        assert_eq!(invalid_cond.status, "False");
+        assert_eq!(invalid_cond.reason, "NoMatchingParent");
+
+        // Find status for valid parent and extract the condition
+        let valid_cond = parent_status
+            .iter()
+            .find(|route_status| route_status.parent_ref.name == "test-valid-server")
+            .expect("must have a route status set for valid server")
+            .conditions
+            .clone()
+            .into_iter()
+            .find(|cond| cond.type_ == "Accepted")
+            .expect("must have at least one 'Accepted' condition set for valid parent");
+        assert_eq!(valid_cond.status, "True");
+        assert_eq!(valid_cond.reason, "Accepted")
+    })
+    .await
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn inbound_no_parent_ref_patch() {
+    with_temp_ns(|client, ns| async move {
+        // A route may not include any parent references. When that's the case,
+        // we expect the controller to simply ignore it.
+        let _route = create(
+            &client,
+            mk_inbound_route(&ns, "test-no-parent-refs-route", None),
+        )
+        .await;
+
+        // Status may not be set straight away. To account for that, wrap a
+        // status condition watcher in a timeout.
+        let api = kube::Api::namespaced(client.clone(), &ns);
+        let resolved = tokio::time::timeout(
+            // 5 seconds should be enough for a route to be patched
+            tokio::time::Duration::from_secs(5),
+            kube::runtime::wait::await_condition(
+                api,
+                "test-no-parent-refs-route",
+                |obj: Option<&k8s::policy::HttpRoute>| -> bool {
+                    obj.and_then(|route| route.status.as_ref()).is_some()
+                },
+            ),
+        )
+        .await;
+
+        // If timeout has elapsed, then route did not receive a status patch
+        assert!(
+            resolved.is_err(),
+            "status condition should timeout when no parent refs are specified"
+        );
+    })
+    .await
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn inbound_accepted_no_parent() {
```

Comment:
Seems like this test can be obviated given the same thing is tested at the beginning of the next one?

Reply:
Hmmm, think you're right. Just need to move the asserts 🤔

Reply:
Oh, yeah! Hadn't even realised...it does seem to be redundant.